servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -494,8 +494,8 @@ private static StreamingHttpConnectionFilterFactory appendConnectionFilter(
     }
 
     @Override
-    public DefaultSingleAddressHttpClientBuilder<U, R> disableHostHeaderFallback() {
-        addHostHeaderFallbackFilter = false;
+    public SingleAddressHttpClientBuilder<U, R> hostHeaderFallback(final boolean enable) {
```

Comment:
Why the return type have changed? Consider being consistent with all other methods that return `DefaultSingleAddressHttpClientBuilder`. If you prefer to keep `SingleAddressHttpClientBuilder`, consider doing that in a follow-up (when you convert abstract classes into interfaces) for all methods.

Reply:
I didn't intend to change anything. As a matter of fact, this is a different, newly added method. However, the previous method now retains the parent return type, as it's not overridden here. This implementation of the new method was auto generated by IDE, I didn't put much thought into it. Your comment inspired me to look at other methods in this class. Unfortunately I don't see any consistency, e.g.:

- `public SingleAddressHttpClientBuilder<U, R> executionStrategy(final HttpExecutionStrategy strategy)`
- `public DefaultSingleAddressHttpClientBuilder<U, R> executor(final Executor executor)`

I believe that:
- because the class is package private (so users can't see the actual return types as declared here),
- and we might want to provide pre-baked builders with calls to methods that could be specific to this particular builder,

the return type should be the actual protected class, not the public superclass (soon to be interface) to avoid the need to cast in such case. Perhaps after the PRs for getting rid of abstract classes in favor of interface we could clean it up or leave for when needed, as it is in fact internal.